### ISSUE TITLE
Tutorial: Kickstart of rhel7 with OSCAP Anaconda Add-on

### ISSUE THREAD

**Issue by iankko** [OpenSCAP/website#54]
Create a tutorial to show basic OSCAP Anaconda Addon usage -- installing PCI-DSS RHEL-7.2 Beta system from scratch.

Use the official SSG OAA PCI-DSS kickstart:
&nbsp; &nbsp; [1] https://github.com/OpenSCAP/scap-security-guide/blob/master/RHEL/7/kickstart/ssg-rhel7-pci-dss-server-with-gui-oaa-ks.cfg

Basic steps performed for the system install (virt-manager will be used as an example, you can choose any other tool of your choose to achieve the same effect):
1) Start virt-manager, click New
2) In the "Create new virtual machine" screen select:
    * "Network install (HTTP, FTP, or NFS)" option
3) In the next screen provide "URL:" to the RHEL-7.2 Beta compose tree, click the "URL Options". In the "Kickstart URL:" specify link to the above mentioned SSG RHEL-7.2 PCI-DSS OAA formatted kickstart, <br/> <br/> <b> Important: The provided kickstart path to work properly in needs to be specified in the raw format, not default HTML one! </b> <br/>
4), Specify <b> 20 </b> GiB as the required storage for the VM guest,
5), Possibly adjust VM name, VM configuration before install, and Network selection,
6), Install this VM (you can take couple of screenshots during the installation). See how the installation is / can be performed in unattended way (doesn't require user interaction),
7), Once the installation has finished, perform the scan of that system against the PCI-DSS profile from the scap-security-guide package. The expected result / output should be just one rule (RPM verify permissions) should be failing. The rest of the rules should pass.

Capture couple of screenshots during this VM install. Provide example, how the OAA kickstart looks like in this case (section between
```
%addon org_fedora_oscap
..
%end (of that section is enough) -- Should you be wondering, yes, those three lines are truly enough!
```
)
Show screenshot of the final HTML report after system is installed.

**Comment by msrubar** [comment]
I'd like to ask you for the review of the [tutorial](http://www-rhopenscap.rhcloud.com/resources/documentation/installing-pci-dss-compliant-rhel-7-2-using-oscap-anaconda-add-on/). The text should be ready. Please also verify that the screenshots are alright because the RHEL-7.2 was used.

**Comment by mpreisler** [comment]
Overall I like the text.

1) I would skip the talk about RHEL 7.2 beta and the URL.

2) I know the PCI-DSS is selected using kickstart commands but I think it would be valuable to change that and use the GUI spoke `Security Policy` to set it up. The kickstart commands can be noted.

Using GUI makes the tutorial easier for newcomers and allows them to select other profiles without having to figure out the profile ID.

3) I was surprised that the screenshots show virt-manager and the host system. This part of the screenshot is the same on all of them. It is not a release blocker but I found it repetitive.

**Comment by jan-cerny** [comment]
@msrubar Review https://docs.google.com/document/d/1AqlJuu3IQWFHkWchVTEWCUoqSjttDFJ1AAtquAv64Pk/edit?usp=sharing

**Comment by vpodzime** [comment]
I was asked by @isimluk to check the [current (draft?) version](https://www-rhopenscap.rhcloud.com/resources/documentation/installing-pci-dss-compliant-rhel-7-2-using-oscap-anaconda-add-on/) and provide some feedback so here it is:

> The important lines in the kickstart file is the installation of the openscap-scanner which is...

This is definitely not an important line in the kickstart. Actually, it doesn't have to be there at all because the addon makes sure that the ``openscap-scanner`` gets installed.

>  ...should be enoughYou can set the...

Missing space.

> ...because it was used during...

*they were used*

Looks good to me otherwise. Good job!

**Comment by msrubar** [comment]
@mpreisler 
1) ok
2) Why would we want it? I thought that the tutorial is meant for admins that want to install a PCI-DSS compliant server from scratch automatically without any clicking configurations. Anyway, if an admin click on the "Security policy" in this [screen](http://www-rhopenscap.rhcloud.com/wp-content/uploads/2015/10/10.png) He can choose some different policy and then hit "begin installation". If he wouldn't hit any button the installation process would continue automatically. I think that the purpose of the tutorial is to show how easily a PCI-DSS compliant server can be installed from scratch automatically.
3) I left it this way because screens looks the same at the page. If I would crop the host system from them some of them would have different size and this may not look nice if you'd look at the page as a whole. Anyway, I'll discuss it with @lhorakov.

**Comment by msrubar** [comment]
@jan-cerny 
Thank you for the review! I went through all you notes and I have a question. Is it alright to say that the failing rule fails because of known issue in the oscap, reference the issue and put a note into the ticket to delete after the issues is solved? Will the fix get to the 7.2?

**Comment by mpreisler** [comment]
> Anyway, if an admin click on the "Security policy" in this screen He can choose some different policy and then hit "begin

Yeah, so maybe we can hint that in the tutorial. I feel that it goes too fast through the screenshots without explaining anything. A small pause and focus on the security spoke might help to bring more context for the user.

**Comment by jan-cerny** [comment]
@msrubar I would not speak about that issue at all.

@mpreisler I agree with you that a security related paragraph would be very helpful for a reader. @msrubar What do you think about it?

**Comment by isimluk** [comment]
I agree with @mpreisler, that showing anaconda gui for choosing policy, would be highly benefitial. However, this tutorial is about using kickstart -> hence the user is not coming on the selection page.

**Comment by mpreisler** [comment]
Another random idea: Let's reference https://www-rhopenscap.rhcloud.com/resources/documentation/make-a-rhel7-server-compliant-with-pci-dss/ and outline the key differences. The two tutorials have a lot in common. We probably want to link from the mentioned tutorial to this tutorial as well.

**Comment by msrubar** [comment]
I updated the information about the openscap-scanner which doesn't have to be in the kickstart file ([SSG issue](https://github.com/OpenSCAP/scap-security-guide/issues/785)). I reassigned it to @lhorakov to check it from the design POV and then we can finally close it.

**Comment by lhorakov** [comment]
If I understood it right, we are talking about this page:
http://www-rhopenscap.rhcloud.com/resources/documentation/make-a-rhel7-server-compliant-with-pci-dss/

If it's so, I redesigned it and for me it's for close.

**Comment by jmercier-ndp** [comment]
Ugh!  No, I too find the lack of clear documentation and examples mind boggling.

I am trying to integrate RHEL7 XCCDF content with a tailoring file into a kickstart file.  But the only available examples that I can find reference datastream.  The documentation makes it unclear which options are manadatory, which are absolute, which are base paths (such as content-url, is it a base path for other options?), nor does it show even cursory examples for each common use case. 

Now that's fine, I can figure it out with trial and error.  But whenever Kickstart bombs systemd hides the error and then reboots!  I am ready to pull my hair out.

So to recap:  I am very good with UNIX and Linux (pre-systemd) and DISA STIG but am confounded on how to integrate oscap into my kickstarts.

I may just skip this step and add an oscap recipe to my Chef post-install.
